### ISSUE TITLE
Catch custom client error

### ISSUE THREAD

**Issue by huang-julien** [nuxt-modules/prismic#223]
### Is your feature request related to a problem? Please describe.

When using a custom client, users can do some async task to retrieve the repository name or for any other reason. If this fails, the whole app will crash.

### Describe the solution you'd like

The prismic plugin should stop if the custom client fails to be resolved. Of course it means that all composables can crash if used but this would prevent the whole app to crash.

**Comment by lihbr** [comment]
Would adding some king of try/catch there suffice in your opinion? https://github.com/nuxt-modules/prismic/blob/master/src/runtime/plugin.ts#L18

**Comment by huang-julien** [comment]
yup, but this would also raise the question of what should the module or prismic do if we fails to resolve the custom client 🤔 . Like nothing and let composables crassh ?

**Comment by lihbr** [comment]
I believe that's better yes, we wouldn't want to add a mock client instead as this would create some more complex issue to debug where you think everything goes well and there's an issue with the client itself where it's actually the user's client factory that fails.
